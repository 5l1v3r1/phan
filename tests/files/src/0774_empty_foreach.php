<?php
function test_foreach() {
    $x = [];
    $y = rand() ? [] : null;
    $z = rand() ? [] : STDIN;
    foreach ($x as $a) {echo $a;}
    foreach ($y as $a) {echo $a;}
    foreach ($z as $a) {echo $a;}
    foreach ([] as $a) {echo $a;}
}
