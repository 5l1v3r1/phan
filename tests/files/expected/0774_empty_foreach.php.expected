%s:6 PhanEmptyForeach Saw a foreach statement with empty iterable type array{}
%s:7 PhanEmptyForeach Saw a foreach statement with empty iterable type array{}|null
%s:8 PhanEmptyForeach Saw a foreach statement with empty iterable type array{}|resource
%s:9 PhanEmptyForeach Saw a foreach statement with empty iterable type array{}
